widgets/executionLogs/src/LogsTable.tsx

```diff
@@ -1,37 +1,74 @@
 import { isEmpty } from 'lodash';
+import { useEffect } from 'react';
 import LogMessage from './LogMessage';
 import type { Event, ExecutionLogsData } from './types';
+import { basePageSize, translate } from './consts';
 
 interface LogsTableProps {
-    data: ExecutionLogsData;
+    items: ExecutionLogsData['items'];
+    executionId: string;
+    moreItemsAvailable: boolean;
+    pageSize: number;
+    toolbox: Stage.Types.Toolbox;
 }
-export default function LogsTable({ data }: LogsTableProps) {
-    const { Icon, Table } = Stage.Basic;
+export default function LogsTable({ items, executionId, moreItemsAvailable, pageSize, toolbox }: LogsTableProps) {
+    const { Button, Icon, Table } = Stage.Basic;
     const { ErrorCausesModal, Utils } = Stage.Common.Events;
     const { Json } = Stage.Utils;
     const { useResettableState } = Stage.Hooks;
     const [event, setEvent, resetEvent] = useResettableState<Event | undefined>(undefined);
 
+    const setPageSize = (size: number) => toolbox.refresh({ gridParams: { pageSize: size } });
+    const showMoreLogs = () => setPageSize(pageSize + basePageSize);
+
+    useEffect(() => {
+        setPageSize(basePageSize);
+    }, [executionId]);
+
     return (
         <>
             <Table basic color="black" compact inverted>
+                {!moreItemsAvailable && (
+                    <Table.Header>
+                        <Table.Row>
+                            <Table.HeaderCell colspan={3} textAlign="center">
+                                <Button
+                                    basic
+                                    compact
+                                    content={translate('buttons.showMoreLogs')}
+                                    inverted
+                                    onClick={showMoreLogs}
+                                />
+                            </Table.HeaderCell>
+                        </Table.Row>
+                    </Table.Header>
+                )}
                 <Table.Body>
-                    {data.items.map(item => {
+                    {items.map(item => {
                         const messageText = Json.stringify(item.message, false);
                         const isErrorLog = Utils.isError(item);
                         const hasErrorCauses = !isEmpty(item.error_causes);
                         const timestamp = Utils.getFormattedTimestamp(item);
 
                         return (
-                            // eslint-disable-next-line no-underscore-dangle
-                            <Table.Row key={item._storage_id} error={isErrorLog} verticalAlign="top">
+                            <Table.Row
+                                // eslint-disable-next-line no-underscore-dangle
+                                key={`${item._storage_id} + ${item.reported_timestamp}`}
```

Comment:
Have you had in mind:
```suggestion
                                key={item._storage_id + item.reported_timestamp}
```
? 🤔 

I was looking for similar patterns across our codebase and it seems that from most of the time we were doing `{param1_param2}` but I've found also examples of doing `{numberParam1+numberParam2}`

That's why I wanted to ask about it as string literal with a plus in the middle seems like a unconventional `key` prop value 😁

I'm not against it, but just wanted to ask about it

Reply:
Good catch. I'll change it according to the suggestion.